src/wp-includes/blocks.php

```diff
@@ -208,43 +208,50 @@ function register_block_style_handle( $metadata, $field_name, $index = 0 ) {
 	}
 
 	// Check whether styles should have a ".min" suffix or not.
-	$suffix    = SCRIPT_DEBUG ? '' : '.min';
-	$style_uri = plugins_url( $style_path, $metadata['file'] );
+	$suffix = SCRIPT_DEBUG ? '' : '.min';
 	if ( $is_core_block ) {
 		$style_path = "style$suffix.css";
-		$style_uri  = includes_url( 'blocks/' . str_replace( 'core/', '', $metadata['name'] ) . "/style$suffix.css" );
 	}
-
 	$style_path_norm = wp_normalize_path( realpath( dirname( $metadata['file'] ) . '/' . $style_path ) );
-	$is_theme_block  = 0 === strpos( $style_path_norm, $theme_path_norm );
-
-	if ( $is_theme_block ) {
-		$style_uri = get_theme_file_uri( str_replace( $theme_path_norm, '', $style_path_norm ) );
+	$has_style_file  = '' !== $style_path_norm;
+	if ( $has_style_file ) {
+		$style_uri      = plugins_url( $style_path, $metadata['file'] );
+		$is_theme_block = 0 === strpos( $style_path_norm, $theme_path_norm );
```

Comment:
Here, `str_starts_with()` could/should be used.

Reply:
I wasn't aware we polyfill now `str_starts_with`. I'm happy to update 👍🏻